src/Symfony/Component/Console/Command/InterruptibleTrait.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Exception\RuntimeException;
+
+/**
+ * This trait adds ability to interrupt safely long-running or endless commands in case "^C" is pressed or "$ kill -2".
+ *
+ * @author Maksym Slesarenko <maks.slesarenko@gmail.com>
+ *
+ * @example
+ * protected function execute(InputInterface $input, OutputInterface $output)
+ * {
+ *     while (true) {
+ *         $this->isInterrupted(true);
+ *
+ *         //doSomething
+ *         sleep(100);
+ *    }
+ *    // or
+ *    while (!$this->isInterrupted()) {
+ *         //doSomething
+ *         sleep(100);
+ *    }
+ * }
+ */
+trait InterruptibleTrait
+{
+    private $isInterrupted = false;
+
+    public function __construct($name = null)
```

Comment:
How is it going to play if the user overrides its command's constructor? Do we need to have these `pcntl_signal`
 calls in another function so it can be called by the command when ran? 🤔